### ISSUE TITLE
Potentially misleading error message if invalid column specified in a JOIN ("Key…is not set correctly")

### ISSUE THREAD

**Issue by rmoff** [confluentinc/ksql#1319]
In this query, the stream `interaction` doesn't have a column called `bookid`: 

```
ksql> select b.title from interaction i left join book b on i.bookId = b.bookId;
 Line: 1, Col: 64 : Invalid join criteria (I.BOOKID = B.BOOKID). Key for I is not set correctly.
```

Given the confusion around table keys in joins (#749 etc) it would be good to make this message much clearer. For example

    Line: 1, Col: 64 : Invalid join criteria (I.BOOKID = B.BOOKID). Column `I.BOOKID` does not exist.

**Comment by rmoff** [comment]
+1 this bit me again today. We should fix the error message to spell out that a column does not exist.

**Comment by GHGautamGH** [comment]
Hi,

I am getting the same error with JOINs when executing from a script file -- either as RUN SCRIPT from the CLI or as --queries-file with command line. The join runs fine interactively. 

My version is 4.1.1. Should I upgrade to 5.0? Has it been fixed on 5.0?

**Comment by rmoff** [comment]
This has improved in 5.1 but could still be better :) 

## Schema

```
ksql> describe users;

Name                 : USERS
 Field        | Type
------------------------------------------
 ROWTIME      | BIGINT           (system)
 ROWKEY       | VARCHAR(STRING)  (system)
 REGISTERTIME | BIGINT
 USERID       | VARCHAR(STRING)
 REGIONID     | VARCHAR(STRING)
 GENDER       | VARCHAR(STRING)
------------------------------------------
For runtime statistics and query details run: DESCRIBE EXTENDED <Stream,Table>;
ksql> describe pageviews;

Name                 : PAGEVIEWS
 Field    | Type
--------------------------------------
 ROWTIME  | BIGINT           (system)
 ROWKEY   | VARCHAR(STRING)  (system)
 VIEWTIME | BIGINT
 USERID   | VARCHAR(STRING)
 PAGEID   | VARCHAR(STRING)
--------------------------------------
For runtime statistics and query details run: DESCRIBE EXTENDED <Stream,Table>;
```

## Valid join

```
ksql> select v.pageid, u.regionid from pageviews v left join users u on v.userid = u.userid limit 1;
Page_83 | Region_2
Limit Reached
Query terminated
```

## Invalid join 1

```
ksql> select v.pageid, u.regionid from pageviews v left join users u on v.userid = u.foobar;
 Line: 1, Col: 76 : Invalid join criteria (V.USERID = U.FOOBAR). Could not find a join criteria operand for U.
ksql>
```

> Could not find a join criteria operand for U.

This is not as precise as it could be. Better would be something like: 

> Column `u.foobar` does not exist.

## Invalid join 2

```
ksql> select v.pageid, u.regionid from pageviews v left join users u on foobar = u.userid;
Failed to prepare statement: Field FOOBAR is ambiguous.
Caused by: Field FOOBAR is ambiguous.
```

This is ok - foobar is ambiguous, it's not been qualified. 

## Invalid join 3

```
ksql> select v.pageid, u.regionid from pageviews v left join foobar_users u on v.userid = u.userid;
Failed to prepare statement: FOOBAR_USERS does not exist.
Caused by: FOOBAR_USERS does not exist.
ksql>
```

This is ok - `Caused by: FOOBAR_USERS does not exist.` is exactly right.
